src/binaries/query/main.rs

```diff
@@ -63,9 +63,11 @@ async fn main_entrypoint(mem_tracker: Arc<MemoryTracker>) -> Result<()> {
     init_default_metrics_recorder();
     set_panic_hook();
 
-    let size = conf.query.max_memory_usage as i64;
-    info!("Set memory limit: {}", size);
-    mem_tracker.set_limit(size);
+    if conf.query.max_memory_limit_enabled {
+        let size = conf.query.max_memory_usage as i64;
```

Comment:
why not use `max_memory_usage = 0` instead.

Reply:
The max_memory_usage is referenced otherwhere.